### ISSUE TITLE
Suggest str methods for trivial regexes

### ISSUE THREAD

**Issue by killercup** [Manishearth/rust-clippy#596]
The `contains` and `{starts,ends}_with` methods of `str` can solve many simple regex cases.

cf. #587

**Comment by llogiq** [comment]
As with the regex validation lint, the trick is using `const_eval::eval_const_partial` on the regex compile argument, and check for zero or one ^ at the beginning, zero or one $ at the end and characters without special regex meaning in between.

**Comment by llogiq** [comment]
Also note that the following holds:

Start|End|String method/op
-----|---|------------
^     |$  |==
       |$  |.ends_with(_)
^    |    |.starts_with(_)
     |   |.contains(_)

**Comment by llogiq** [comment]
We could add suggestions taking the original str (beware of backslashes, though).

**Comment by llogiq** [comment]
Speaking of backslashes, I found the following false negative:

```
    let trivial_backslash = Regex::new("a\\.b");
    //~^ERROR: trivial regex
    //~|HELP consider using `str::contains`
```

**Comment by mcarton** [comment]
@llogiq I made a PR to fix that false negative. What kind of suggestion do you propose?

**Comment by llogiq** [comment]
This would require using an ExprUseVisitor to find all usages of the regex and suggest using `.contains("a.b")` instead.
